app/Http/Controllers/WebAuthn/WebAuthnManageController.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace App\Http\Controllers\WebAuthn;
+
+use App\Exceptions\UnauthenticatedException;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+
+class WebAuthnManageController
+{
+	/**
+	 * @throws UnauthenticatedException
+	 */
+	public function list(): Collection
+	{
+		/** @var \App\Models\User $user */
+		$user = Auth::user() ?? throw new UnauthenticatedException();
+
+		return $user->webAuthnCredentials;
+	}
+
+	/**
+	 * @throws UnauthenticatedException
+	 */
+	public function delete(Request $request): void
+	{
+		$id = $request->validate(['id' => 'required|string']);
+
+		/** @var \App\Models\User $user */
+		$user = Auth::user() ?? throw new UnauthenticatedException();
+		$user->webAuthnCredentials()->where('id', $id)->delete();
+	}
```

Comment:
In order to follow the behaviour authenticate first or throw 401, validate request or throw 422 and finally authorize or throw 403, the order of checks must be swapped. Or create a proper request class for this method as argument.

```suggestion
	public function delete(Request $request): void
	{
		/** @var \App\Models\User $user */
		$user = Auth::user() ?? throw new UnauthenticatedException();
		$id = $request->validate(['id' => 'required|string']);
		$user->webAuthnCredentials()->where('id', $id)->delete();
	}
```

Reply:
fixed [d2052dc](https://github.com/LycheeOrg/Lychee/pull/1469/commits/d2052dc2dbd87fe49b1bf75434ef40ed6ce22e29)